src/Features/CSharp/Portable/CodeFixes/Nullable/CSharpDeclareAsNullableCodeFixProvider.cs

```diff
@@ -0,0 +1,155 @@
+        // warning CS8600: Converting null literal or possible null value to non-nullable type.
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create("CS8625", "CS8600");
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var diagnostic = context.Diagnostics.First();
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var node = root.FindNode(diagnostic.Location.SourceSpan, getInnermostNodeForTie: true);
+
+            var declarationTypeToFix = TryGetDeclarationTypeToFix(node);
+            if (declarationTypeToFix == null)
+            {
+                return;
+            }
+
+            context.RegisterCodeFix(new MyCodeAction(
+                c => FixAsync(context.Document, diagnostic, c)),
+                context.Diagnostics);
+        }
+
+        protected override Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var root = editor.OriginalRoot;
+
+            // a method can have multiple `return null;` statements, but we should only fix its return type once
+            var alreadyHandled = PooledHashSet<TypeSyntax>.GetInstance();
+
+            foreach (var diagnostic in diagnostics)
+            {
+                var node = diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken);
+                MakeDeclarationNullable(document, editor, node, alreadyHandled);
+            }
+
+            alreadyHandled.Free();
+            return Task.CompletedTask;
+        }
+
+        private static void MakeDeclarationNullable(Document document, SyntaxEditor editor, SyntaxNode node, HashSet<TypeSyntax> alreadyHandled)
+        {
+            var declarationTypeToFix = TryGetDeclarationTypeToFix(node);
+            if (declarationTypeToFix != null && alreadyHandled.Add(declarationTypeToFix))
+            {
+                var fixedDeclaration = SyntaxFactory.NullableType(declarationTypeToFix.WithoutTrivia()).WithTriviaFrom(declarationTypeToFix);
+                editor.ReplaceNode(declarationTypeToFix, fixedDeclaration);
+            }
+        }
+
+        private static TypeSyntax TryGetDeclarationTypeToFix(SyntaxNode node)
+        {
+            if (!node.IsKind(SyntaxKind.NullLiteralExpression))
+            {
+                return null;
+            }
+
+            if (node.IsParentKind(SyntaxKind.ReturnStatement))
+            {
+                var containingMember = node.GetAncestors().FirstOrDefault(a => a.IsKind(SyntaxKind.MethodDeclaration, SyntaxKind.PropertyDeclaration,
+                    SyntaxKind.ParenthesizedLambdaExpression, SyntaxKind.SimpleLambdaExpression, SyntaxKind.LocalFunctionStatement));
```

Comment:
❕ Also need `AnonymousMethodExpression`

Reply:
Good catch. Thanks

Reply:
I think you could use `FirstOrDefault(IsReturnableConstruct)`

Reply:
also, isn't there an overload of `GetAncestorOrThis` or `FirstAncestorOrSelf` that takes a predicate somewhere?

Reply:
> IsReturnableConstruct

Even better. Thanks!

> isn't there a method like ... that takes a predicate somewhere?

Didn't find one. I'll leave as-is.

Reply:
Actually... that might be wrong. IsReturnableConstruct considers individual accessors whereas you're looking at the actual property. I don't think this will work. Sorry :/